### DISCUSSION TITLE
Let the apps open links

### DISCUSSION THREAD

**Discussion by taorepoara** [lenra-io/lenra_cli#174] [Polls]
@lenra-io/all  We shoud add the possibility to the apps to open URLs in a browser or a WebView, but we should secure it by asking the user if he want to continue.

I see many ways to handle it:

1. Create a specific component to handle it
2. Create a specific listener interpreted by the client
3. Create an API endpoint to be called from an app listener

To let the user keep the control on it, we will ask him if he want to open the link or not (like Discord does) with many options (specific to the current app): 
- The answer:
  - Yes
  - No
- Duration: Don't ask again for 
  - the session
  - today
  - a week
  - etc...
- Repetition target: Don't ask again for 
  - This specific link
  - This domain
  - Every links

The first solution is the easiest one, but it adds a component just for that...

The second one let us manage this  from more events, adds a behavior not managed yet. We should think first if it could be useful for something else.

The third one let the dev have more control on it (that could be good or not), but add problems (ex: if the user is connected from many devices, do we open it on every devices)

**Comment by taorepoara** [comment]
We also could mix the solutions 2 and 3 to create a custom listener that could also be called from an app listener.

**Comment by jonas-martinez** [comment]
@taorepoara This can now be done using a lenra listener in the application. We could create a `@lenra:link` reserved listener that handles the generation of a short URL that redirects to a lenra.io page asking if the user wants to continue on the link.

```json
{
  "type": "button",
  "text": "Go to Google",
  "onPressed": {
    "action": "@lenra:link",
    "props": {
      "link": "google.com"
    }
  }
}
```

This should be really easy to implement.

**Reply by taorepoara** [reply]
I think that the existing navTo could do the trick
